src/components/SearchBarPopup/SearchBarPopup.tsx

```diff
@@ -0,0 +1,159 @@
+                {id: 3, name: "Place 3", address: "City, County", isAdded: false }];
+            resolve(results);
+        })
+            .then((results: Destination[]) => {
+                this.setState({ resultsOpen: true, loading: false, searchKey: searchInput });
+
+                // Todo: this needs to be restored afterwards! 
+                // this.setState({ resultsOpen: true, results: results, loading: false });
+            })
+            .catch((err) => {
+                this.setState({
+                    resultsOpen: false,
+                    loading: false,
+                    errMsg: err.toString(),
+                });
+            });
+    }
+
+    private addRemovePlace (id: number) {
+        const elementsIndex = this.state.results.findIndex(element => element.id === id);
+        let newResults = [...this.state.results];
+        newResults[elementsIndex] = { ...newResults[elementsIndex], isAdded: !newResults[elementsIndex].isAdded}
+        this.setState({
+            results: newResults,
+            resultsOpen: true
+        })
+    }
+
+    componentDidMount(): void {
+        document.addEventListener("mousedown", this.handleClickOutside, false);
+    }
+
+    componentWillUnmount(): void {
+        document.removeEventListener("mousedown", this.handleClickOutside, false);
+    }
+
+    private handleClickOutside = (event: MouseEvent) => {
+        if (this.container.current && event.target instanceof Node) {
+            if (this.container.current.contains(event.target)) {
+                return;
+            }  else {
+                // TODO: backend, send the updated result list to the backend
+            }
+        }
+    };
+    
+    private renderSearchResults() {
+        const { results } = this.state;
+        // console.log(this.state.resultsOpen)
+        if (this.state.resultsOpen && _.isArray(results) && !_.isEmpty(results)) {
+            return (
+                <div className="results-container">
+                    <ul className="results-list">
+                        {results.map((result) => (
+                            // temporarily using result as key. Change to some id later
+                            // TODO: replace with valid href
+                            // TODO:<a href> remount the entire component, fix when connecting to backend.
+                            <li className="results-entry" key={result["name"]} onClick={() =>this.addRemovePlace(result["id"])}>
+                                <a href="placeholder">
+                                    <span className="placeName">{result["name"]}</span>
+                                    <span className="address">{result["address"]}</span>
+                                </a>
+                                <a href="placeholder">                                    
+                                <FontAwesomeIcon className={result["isAdded"] ? "checked" : "checked icon-hidden"} icon="check" fixedWidth /></a>
```

Comment:
Instead of using className to handle icon hiding, suggest doing this:
```suggestion
                                {result.isAdded &&
                                    <FontAwesomeIcon icon="check" />
                                }
```
This will only render the <FontAwesomeIcon> element if result.isAdded is true.
Also removed the fixedWidth, this is more for when we have multiple icons lined vertically and we wanna make sure they all have same width for alignment purposes.